File: Prixly/Prixly.Dal/Models/CreditCardDto.cs
namespace Prixly.Dal.Models
{
    public class CreditCardDto
    {
        public Guid Guid { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public Guid BankGuid { get; set; }

        public string ImageUrl { get; set; } = string.Empty;

        public bool IsActive { get; set; } = true;

        public List<CustomRewardRuleDto> CustomRewardRules { get; set; } = new List<CustomRewardRuleDto>();
        public List<StandardRewardRuleDto> StandardRewardRules { get; set; } = new List<StandardRewardRuleDto>();
    }
}

File: Prixly/Prixly.Client/src/models/credit-card-dto.ts
import { CustomRewardRuleDto } from "./custom-reward-rule-dto";
import { StandardRewardRuleDto } from "./standard-reward-rule-dto";

export interface CreditCardDto {
  guid: string;
  name: string;
  description: string;
  bankGuid: string;
  imageUrl: string;
  isActive: boolean;
  customRewardRules: CustomRewardRuleDto[];
  standardRewardRules: StandardRewardRuleDto[];
}
